internal/services/mssql/mssql_server_dns_alias_resource.go

```diff
@@ -0,0 +1,143 @@
+package mssql
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/mssql/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/mssql/validate"
+	webValidate "github.com/hashicorp/terraform-provider-azurerm/internal/services/web/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type ServerDNSAliasModel struct {
+	MsSQLServerId string `tfschema:"mssql_server_id"`
+	Name          string `tfschema:"name"`
+	DNSRecord     string `tfschema:"dns_record"`
+}
+
+type ServerDNSAliasResource struct{}
+
+var _ sdk.Resource = (*ServerDNSAliasResource)(nil)
+
+func (m ServerDNSAliasResource) Arguments() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"mssql_server_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.ServerID,
+		},
+
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: webValidate.AppServiceEnvironmentName,
+		},
+	}
+}
+
+func (m ServerDNSAliasResource) Attributes() map[string]*schema.Schema {
+	return map[string]*pluginsdk.Schema{
+		"dns_record": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+	}
+}
+
+func (m ServerDNSAliasResource) ModelObject() interface{} {
+	return &ServerDNSAliasModel{}
+}
+
+func (m ServerDNSAliasResource) ResourceType() string {
+	return "azurerm_mssql_server_dns_alias"
+}
+
+func (m ServerDNSAliasResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			var alias ServerDNSAliasModel
+			if err := metadata.Decode(&alias); err != nil {
+				return err
+			}
+			serverID, err := parse.ServerID(alias.MsSQLServerId)
+			if err != nil {
+				return err
+			}
+			client := metadata.Client.MSSQL.ServerDNSAliasClient
+			subscriptionID := metadata.Client.Account.SubscriptionId
+			id := parse.NewServerDNSAliasID(subscriptionID, serverID.ResourceGroup, serverID.Name, alias.Name)
```

Comment:
Can we do this instead?
```suggestion
			id := parse.NewServerDNSAliasID(serverID.SubscriptionId, serverID.ResourceGroup, serverID.Name, alias.Name)
```